server/tests/ballot_comparison/test_ballot_comparison.py

```diff
@@ -319,6 +319,116 @@ def test_contest_names_dont_match_cvr_contests(
     }
 
 
+def audit_all_ballots(
+    round_id: str, audit_results, target_contest_id, opportunistic_contest_id
+):
+    choice_1_1, choice_1_2 = Contest.query.get(target_contest_id).choices
+    choice_2_1, choice_2_2, choice_2_3 = Contest.query.get(
+        opportunistic_contest_id
+    ).choices
+
+    def ballot_key(ballot: SampledBallot):
+        return (
+            ballot.batch.jurisdiction.name,
+            ballot.batch.tabulator,
+            ballot.batch.name,
+            ballot.ballot_position,
+        )
+
+    round = Round.query.get(round_id)
+    sampled_ballots = (
+        SampledBallot.query.filter_by(status=BallotStatus.NOT_AUDITED)
+        .join(SampledBallotDraw)
+        .filter_by(round_id=round.id)
+        .join(Batch)
+        .order_by(Batch.tabulator, Batch.name, SampledBallot.ballot_position)
+        .all()
+    )
+    sampled_ballot_keys = [ballot_key(ballot) for ballot in sampled_ballots]
+
+    assert sorted(sampled_ballot_keys) == sorted(list(audit_results.keys()))
+
+    for ballot in sampled_ballots:
+        interpretation_str, _ = audit_results[ballot_key(ballot)]
+
+        if interpretation_str == "not found":
+            ballot.status = BallotStatus.NOT_FOUND
+            continue
+
+        ballot.status = BallotStatus.AUDITED
+
+        if interpretation_str == "blank":
+            audit_ballot(ballot, target_contest_id, Interpretation.BLANK)
+            audit_ballot(ballot, opportunistic_contest_id, Interpretation.BLANK)
+
+        else:
+            (
+                vote_choice_1_1,
+                vote_choice_1_2,
+                vote_choice_2_1,
+                vote_choice_2_2,
+                vote_choice_2_3,
+            ) = interpretation_str.split(",")
+
+            target_choices = ([choice_1_1] if vote_choice_1_1 == "1" else []) + (
+                [choice_1_2] if vote_choice_1_2 == "1" else []
+            )
+            audit_ballot(
+                ballot,
+                target_contest_id,
+                (
+                    Interpretation.VOTE
+                    if vote_choice_1_1 != ""
+                    else Interpretation.CONTEST_NOT_ON_BALLOT
```

Comment:
This feels weird to me. We're saying that a vote for the loser is the same as the contest not being on the ballot?

Reply:
this line is just interpreting test data for instructions on how to audit the ballots. it's copied from what we already had in the ballot comparison tests. it just says that if we put an empty string in the test data, audit that as "contest not on ballot"

Reply:
ah I gotcha